planet/cli/data.py

```diff
@@ -23,83 +45,74 @@ def parse_item_types(ctx, param, value: str) -> List[str]:
 
 def parse_filter(ctx, param, value: str) -> dict:
     """Turn filter JSON into a dict."""
-    return json.loads(value)
-
-
-@click.group()
-@click.pass_context
-def data(ctx):
-    """A group of commands for interacting with the Data API."""
+    # read filter using raw json
+    if value.startswith('{'):
+        try:
+            json_value = json.loads(value)
+        except json.decoder.JSONDecodeError:
+            raise click.BadParameter('Filter does not contain valid json.',
+                                     ctx=ctx,
+                                     param=param)
+        if json_value == {}:
+            raise click.BadParameter('Filter does not contain valid json.',
+                                     ctx=ctx,
+                                     param=param)
+        return json_value
+    # read filter using click pipe option
+    elif value == '-' or Path(value).name.endswith('.json'):
```

Comment:
@mkshah605 @kevinlacaille we can't rely on file extensions. Some people use .json, some use .geojson, some use no extension at all.

I think `else:` would be fine.